mypy/semanal.py

```diff
@@ -2089,6 +2007,105 @@ def analyze_lvalue(self, lval: Lvalue, nested: bool = False,
+        """
+        # Top-level definitions within some statements (at least while) are
+        # not handled in the first pass, so they have to be added now.
+        nested_global = (not self.is_func_scope() and
+                         self.block_depth[-1] > 0 and
+                         not self.type)
+        if (add_global or nested_global) and lval.name not in self.globals:
+            # Define new global name.
+            v = self.make_name_lvalue_var(lval, GDEF)
+            self.globals[lval.name] = SymbolTableNode(GDEF, v)
+        elif isinstance(lval.node, Var) and lval.is_new_def:
+            if lval.kind == GDEF:
+                # Since the is_new_def flag is set, this must have been analyzed
+                # already in the first pass and added to the symbol table.
+                # An exception is typing module with incomplete test fixtures.
+                assert lval.node.name() in self.globals or self.cur_mod_id == 'typing'
+        elif (self.locals[-1] is not None and lval.name not in self.locals[-1] and
+              lval.name not in self.global_decls[-1] and
+              lval.name not in self.nonlocal_decls[-1]):
+            # Define new local name.
+            v = self.make_name_lvalue_var(lval, LDEF)
+            self.add_local(v, lval)
+            if lval.name == '_':
+                # Special case for assignment to local named '_': always infer 'Any'.
+                typ = AnyType(TypeOfAny.special_form)
+                self.store_declared_types(lval, typ)
+        elif not self.is_func_scope() and (self.type and
+                                           lval.name not in self.type.names):
+            # Define a new attribute within class body.
+            v = self.make_name_lvalue_var(lval, MDEF)
+            v.is_inferred = not explicit_type
+            self.type.names[lval.name] = SymbolTableNode(MDEF, v)
+        else:
+            self.make_name_lvalue_point_to_existing_def(lval, explicit_type, final_cb)
+
+    def make_name_lvalue_var(self, lvalue: NameExpr, kind: int) -> Var:
+        """Return a Var node for an lvalue that is a name expression."""
+        v = Var(lvalue.name)
+        v.set_line(lvalue)
+        if kind == MDEF:
+            assert self.type is not None
+            v.info = self.type
+            v.is_initialized_in_class = True
+        if kind != LDEF:
+            v._fullname = self.qualified_name(lvalue.name)
+        if kind == GDEF:
+            v.is_ready = False  # Type not inferred yet
+        lvalue.node = v
+        lvalue.is_new_def = True
+        lvalue.is_inferred_def = True
+        lvalue.kind = kind
+        if kind == GDEF:
+            lvalue.fullname = v._fullname
+        else:
+            lvalue.fullname = lvalue.name
+        return v
+
+    def make_name_lvalue_point_to_existing_def(
+            self,
+            lval: NameExpr,
+            explicit_type: bool,
+            final_cb: Optional[Callable[[bool], None]]) -> None:
```

Comment:
I know this is really minor, but I would add a docstring here (jus point to `analyze_lvalue` for args, like you did above).

Reply:
OK, this is not essential. I am going to merge this just to reduce number of open PRs